pkg/clients/iam/role.go

```diff
@@ -175,13 +175,19 @@ func IsRoleUpToDate(in v1beta1.RoleParameters, observed iamtypes.Role) (bool, st
 		return false, "", err
 	}
 
+	ignoreFields := []string{"AssumeRolePolicyDocument"}
 	policyUpToDate, err := isAssumeRolePolicyUpToDate(desired.AssumeRolePolicyDocument, observed.AssumeRolePolicyDocument)
 	if err != nil {
 		return false, "", err
 	}
 
-	diff := cmp.Diff(desired, &observed, cmpopts.IgnoreInterfaces(struct{ resource.AttributeReferencer }{}), cmpopts.IgnoreFields(observed, "AssumeRolePolicyDocument"), cmpopts.IgnoreTypes(document.NoSerde{}))
-	if diff == "" && policyUpToDate {
+	_, _, areTagsUpToDate := DiffIAMTagsReturnStringList(desired.Tags, observed.Tags)
+	if areTagsUpToDate {
+		ignoreFields = append(ignoreFields, "Tags")
+	}
+
+	diff := cmp.Diff(desired, &observed, cmpopts.IgnoreInterfaces(struct{ resource.AttributeReferencer }{}), cmpopts.IgnoreFields(observed, ignoreFields...), cmpopts.IgnoreTypes(document.NoSerde{}))
```

Comment:
I believe it is cleaner to use the builtin `cmpopts.SortSlices` function:

```go
cmpopts.SortSlices(lessTag)
```

and

```go
func lessTag(a, b iamtypes.Tag) bool {
	if a.Key == nil {
		return b.Key != nil
	}
	if b.Key == nil {
		return false
	}
	return *a.Key <= *b.Key
}
```